Repository/ProfileRepository.php

```diff
@@ -0,0 +1,359 @@
+            $currentProfile,
+            Profile::TYPE_COMPANION,
+            $limit,
+            $offset,
+            null,
+            null,
+            $sex,
+            $ageFrom,
+            $ageTo,
+            $figureParams,
+            $ethnicity,
+            $orientation,
+            $smoker,
+            $serviceIds,
+            $languageIds,
+            $isPornStar,
+            $escort,
+            $escortToSociety,
+            null,
+            null,
+            null,
+            false,
+            false,
+            null,
+            (bool) $isIndependent,
+            $traveling,
+            $hasVideo,
+            $viewPort,
+            $active,
+            $followed,
+            null,
+            $assignedToProfile,
+            $approved,
+            $bookable
+        );
+
+        if ($orderBy === 'alphabetAsc' || $orderBy === 'alphabetDesc') {
+            $queryBuilder->innerJoin(User::class, 'u', Join::WITH, $this->getAlias() . '.userId = u.id');
+        }
+
+        $queryBuilder = $this->applyOrderBy($queryBuilder, $orderBy, $lat, $lng);
+
+        return $queryBuilder
+            ->getQuery()
+            ->getResult();
+    }
+
+    private function findQueryBuilder(
+        ?Profile $currentProfile,
+        ?string $profileType,
+        ?int $limit,
+        ?int $offset,
+        ?float $lat,
+        ?float $lng,
+        ?array $sex = null,
+        ?int $ageFrom = null,
+        ?int $ageTo = null,
+        ?FigureParams $figureParams = null,
+        ?array $ethnicity = null,
+        ?array $orientation = null,
+        ?array $smoker = null,
+        ?array $serviceIds = null,
+        ?array $languageIds = null,
+        ?bool $isPornStar = null,
+        ?bool $escort = null,
+        ?bool $escortToSociety = null,
+        ?Profile $favoriteProfileByProfile = null,
+        ?array $types = null,
+        ?array $massageTypes = null,
+        bool $isNowOpen = false,
+        bool $isTodayOpen = false,
+        ?bool $isFreeEntry = null,
+        bool $isIndependent = false,
+        ?bool $traveling = null,
+        ?bool $hasVideo = null,
+        ?ViewPort $viewPort = null,
+        ?bool $active = null,
+        ?bool $followed = null,
+        ?bool $recruitment = null,
+        ?Profile $assignToProfile = null,
+        ?bool $approved = true,
+        ?bool $bookable = null,
+    ): QueryBuilder {
+        $queryBuilder = $this->prepareQueryBuilderExtended($limit, $offset, null, $currentProfile, $followed, $recruitment);
+
+        $alias = $this->getAlias();
+        if ($profileType) {
+            $queryBuilder
+                ->andWhere($alias . '.profileType = :profileType')
+                ->setParameter('profileType', $profileType);
+        }
+
+        if ($sex !== null) {
+            $queryBuilder->andWhere($alias . '.sex IN(:sex)')->setParameter('sex', $sex);
+        }
+
+        $currentYear = (new \DateTime())->format('Y');
```

Comment:
Následuje hrozná spousta přidávání where do dotazu. Člověka to hned trkne do oka jak je to dlouhé, ale vlastně mě ani teď nenapadá jak to udělat líp